### PR TITLE
refactor: remove builder coupling from server

### PR DESCRIPTION
**Pull request by pi0**
Changes:
- Move middleware for dev (devMiddleware and hotMiddleware) into `@nuxt/builder:builder.middleware()` and just keep a placeholder inside `@nuxt/server`
- Don't call `nuxt.ready()` in constructor anymore
- Don't create a server during `nuxt build` or when `options.server` is `false`

### CODE REVIEW

**Comment by Atinux** [comment]
😍😍😍

**Comment by galvez** [comment]
Good stuff, I can see where you're going with this :)

**Review by galvez — DISMISSED** [review]

File: packages/webpack/src/builder.js [hunk 1]
```diff
@@ -216,11 +216,22 @@ export class WebpackBundler {
...
+
+  async middleware(req, res, next) {
+    const name = req.modernMode ? 'modern' : 'client'
+
+    if (this.devMiddleware && this.devMiddleware[name]) {
+      await this.devMiddleware[name](req, res)
+    }
+
+    if (this.hotMiddleware && this.hotMiddleware[name]) {
+      await this.hotMiddleware[name](req, res)
```
Comment (manniL) [inline packages/webpack/src/builder.js]: Is the order meaningful here? If not, could we get a perf bonus when using `Promise.all` here?

Comment (pi0) [reply packages/webpack/src/builder.js]: Middleware are sequential by design. Even if it works we are doing an anti pattern. Also only one of them matches a request so the difference. Combining this with the fact that node is single threaded, there should be no difference.

**Comment by clarkdo** [comment]
Did I see Chinese ? @pi0 😄

**Pull request merged**